src/utils.rs

```diff
@@ -28,19 +25,23 @@ pub async fn create_response(
         .await
 }
 
-pub fn get_full_username(user: &User) -> String {
-    format!("{}#{:04}", user.name, user.discriminator)
-}
-
-pub fn get_human_readable_timestamp(duration: Duration) -> String {
-    let seconds = duration.as_secs() % 60;
-    let minutes = (duration.as_secs() / 60) % 60;
-    let hours = duration.as_secs() / 3600;
-
-    if hours < 1 {
-        format!("{:02}:{:02}", minutes, seconds)
+pub fn get_human_readable_timestamp(duration: Option<Duration>) -> String {
+    if let Some(duration) = duration {
+        if duration == Duration::MAX {
+            return "∞".to_string();
+        }
+
+        let seconds = duration.as_secs() % 60;
+        let minutes = (duration.as_secs() / 60) % 60;
+        let hours = duration.as_secs() / 3600;
+
+        if hours < 1 {
+            format!("{:02}:{:02}", minutes, seconds)
+        } else {
+            format!("{}:{:02}:{:02}", hours, minutes, seconds)
+        }
     } else {
-        format!("{}:{:02}:{:02}", hours, minutes, seconds)
+        "∞".to_string()
     }
```

Comment:
This can better be rewritten as a match now that we have three cases: Some duration and infinite, some duration and finite, and the rest